### PR TITLE
fix(DTFS2-7092): code consistency

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure exceptional handling variables are called `error` and `exception` (where error is declared on an upper scope).
Ensure coding pattern is consistent where not followed.

## Resolution :heavy_check_mark:
* Updated variables called `e` to `error` and `exception` respectively
* Improve string substitution usage for code consistency.

## Miscellaneous :heavy_plus_sign:
* Added docstring for functions.
* Changed string substitution formatter to `%o` for enhanced logging.
* Updated `azure/login` GHA plugin to `v2`
* GitHub Actions has stoped supporting `docker-compose` as per their migration plan https://github.com/actions/runner-images/issues/9557. Have upgraded to docker `v2` syntax.

### CODE REVIEW

File: dtfs-central-api/src/index.ts [hunk 1]
```diff
@@ -6,4 +6,4 @@ initialiseCronJobScheduler(cronSchedulerJobs);
 
 const PORT = process.env.PORT || 5005;
 
-app.listen(PORT, () => console.info('✅ Central micro-service initialised on :%s', PORT));
+app.listen(PORT, () => console.info('✅ Central micro-service initialised on $s', PORT));
```
Comment (AlexBramhill) [inline dtfs-central-api/src/index.ts:9]: We sure we want a `$` here?

Comment (abhi-markan) [reply dtfs-central-api/src/index.ts:9]: Good spot, this was accidental have rectified.

File: portal-api/src/v1/users/routes.js [hunk 2]
```diff
@@ -151,8 +151,8 @@ module.exports.updateById = async (req, res, next) => {
         return res.status(200).json(sanitizeUser(updatedUser));
       });
     });
-  } catch (e) {
-    console.error('Error updating user', e);
+  } catch (error) {
+    console.error('Error updating user', error);
```
Comment (AlexBramhill) [inline portal-api/src/v1/users/routes.js:155]: Do we want to inc string here

File: external-api/src/v1/controllers/acbs.controller.ts [hunk 3]
```diff
@@ -109,9 +109,9 @@ const createAcbsRecord = async (deal: any, bank: any) => {
         deal,
         bank,
       },
-    }).catch((e: any) => {
-      console.error('Error creating ACBS record: %O', e);
-      return e;
+    }).catch((error: any) => {
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/src/v1/controllers/acbs.controller.ts:112]: Only `any` can be set to `catch` as Catch does not know what type of object will be thrown.
Using anything other than `any` will result in compilation error.

File: external-api/src/v1/controllers/currency-exchange-rate.controller.ts [hunk 4]
```diff
@@ -58,7 +58,7 @@ export const getExchangeRate = async (req: Request, res: Response) => {
     let sourceCurrency = source;
     let targetCurrency = target;
 
-    console.info('⚡️ Invoking MDM currencies/exchange endpoint: %s:%s', sourceCurrency, targetCurrency);
+    console.info('⚡️ Invoking MDM currencies/exchange endpoint: %s$s', sourceCurrency, targetCurrency);
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/src/v1/controllers/currency-exchange-rate.controller.ts]: Rectified.

File: external-api/src/index.ts [hunk 5]
```diff
@@ -4,4 +4,4 @@ dotenv.config();
 
 const PORT = process.env.PORT || 5002;
 
-app.listen(PORT, () => console.info('✅ External micro-service initialised on :%s', PORT));
+app.listen(PORT, () => console.info('✅ External micro-service initialised on $s', PORT));
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/src/index.ts]: Rectified.

File: gef-ui/server/index.ts [hunk 6]
```diff
@@ -2,4 +2,4 @@ import app from './createApp';
 
 const PORT = process.env.PORT || 5006;
 
-app.listen(PORT, () => console.info('✅ GEF UI micro-service initialised on :%s', PORT));
+app.listen(PORT, () => console.info('✅ GEF UI micro-service initialised on $s', PORT));
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply gef-ui/server/index.ts]: Rectified.

File: portal/server/index.ts [hunk 7]
```diff
@@ -2,4 +2,4 @@ import app from './createApp';
 
 const PORT = process.env.PORT || 5000;
 
-app.listen(PORT, () => console.info('✅ Portal UI micro-service initialised on :%s', PORT));
+app.listen(PORT, () => console.info('✅ Portal UI micro-service initialised on $s', PORT));
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal/server/index.ts]: Rectified.

File: portal-api/src/v1/users/login.controller.js [hunk 8]
```diff
@@ -24,17 +33,17 @@ module.exports.login = (username, password, userService) =>
 
       try {
         userService.validateUserIsActiveAndNotDisabled(user);
-      } catch (e) {
-        console.error('Error logging in: %O', e);
-        if (e instanceof UserBlockedError) {
+      } catch (exception) {
+        console.error('Error logging in: %O', exception);
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/users/login.controller.js:37]: Good spot have updated.

File: portal-api/src/v1/users/routes.js (hunk 2 above)
Comment (abhi-markan) [reply portal-api/src/v1/users/routes.js:155]: Good spot, have updated.

File: portal-api/src/v1/api.js [hunk 9]
```diff
@@ -289,17 +289,17 @@ const getUtilisationReports = async (bankId, options) => {
 
   try {
     if (!isValidBankId(bankId)) {
-      console.error('Get utilisation reports failed with the following bank ID: %s', bankId);
-      throw new Error('Invalid bank ID provided: %s', bankId);
+      console.error('Get utilisation reports failed with the following bank ID %s', bankId);
+      throw new Error('Invalid bank ID provided %s', bankId);
```
Comment (AlexBramhill) [inline portal-api/src/v1/api.js:293]: I'm not sure this works with errors

Comment (abhi-markan) [reply portal-api/src/v1/api.js:293]: Good spot, have rectified.

File: portal-api/src/v1/users/sign-in-link.service.js [hunk 10]
```diff
@@ -141,13 +177,17 @@ class SignInLinkService {
         signInLink,
         signInLinkDuration: `${SIGN_IN_LINK.DURATION_MINUTES} minute${SIGN_IN_LINK.DURATION_MINUTES === 1 ? '' : 's'}`,
       });
-    } catch (e) {
-      const error = new Error('Failed to email the sign in token.');
-      error.cause = e;
-      throw error;
+    } catch (error) {
+      throw new Error('Failed to email the sign in token %o', { cause: error });
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/users/sign-in-link.service.js:181]: Good spot, have rectified.

File: portal-api/src/v1/users/sign-in-link.service.js [hunk 11]
```diff
@@ -122,16 +155,19 @@ class SignInLinkService {
         signInTokenHash: hash,
         expiry,
       });
-    } catch (e) {
-      const error = new Error('Failed to save the sign in token.');
-      error.cause = e;
-      throw error;
+    } catch (error) {
+      throw new Error('Failed to save the sign in token %o', { cause: error });
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/users/sign-in-link.service.js:159]: 1. Good spot, have rectified.

File: portal-api/src/v1/users/sign-in-link.service.js [hunk 12]
```diff
@@ -90,28 +101,50 @@ class SignInLinkService {
...
+   */
   #createSignInToken() {
     try {
       return this.#randomGenerator.randomHexString(SIGN_IN_LINK.TOKEN_BYTE_LENGTH);
-    } catch (e) {
-      const error = new Error('Failed to create a sign in token.');
-      error.cause = e;
-      throw error;
+    } catch (error) {
+      throw new Error('Failed to create a sign in token %o', { cause: error });
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/users/sign-in-link.service.js:139]: Good spot, have rectified.

File: utils/mock-data-loader/banks/index.js [hunk 13]
```diff
@@ -157,7 +157,8 @@ const BANKS = [
       { startMonth: 9, endMonth: 11 },
       { startMonth: 12, endMonth: 2 },
     ],
-    isVisibleInTfmUtilisationReports: true,
+    // TODO: Set this to `true` once quarterly reporting is implemented
+    isVisibleInTfmUtilisationReports: false,
```
Comment (AlexBramhill) [inline utils/mock-data-loader/banks/index.js:161]: out of interest do we have a pr for this

File: portal-api/src/v1/api.js [hunk 14]
```diff
@@ -289,17 +289,17 @@ const getUtilisationReports = async (bankId, options) => {
...
-      throw new Error('Invalid bank ID provided: %s', bankId);
+      console.error('Get utilisation reports failed with the following bank ID %s', bankId);
+      throw new Error(`Invalid bank ID provided ${bankId}`);
     }
 
     if (reportPeriod && !isValidReportPeriod(reportPeriod)) {
-      console.error('Get utilisation reports failed with the following report period: %s', reportPeriod);
-      throw new Error('Invalid report period provided: %s', reportPeriod);
+      console.error('Get utilisation reports failed with the following report period %s', reportPeriod);
+      throw new Error('Invalid report period provided %s', reportPeriod);
```
Comment (AlexBramhill) [inline portal-api/src/v1/api.js:298]: Missed one!

File: portal-api/src/v1/users/sign-in-link.service.create-and-email-sign-in-link.test.js [hunk 15]
```diff
@@ -358,8 +358,8 @@ describe('SignInLinkService', () => {
     it('rejects', async () => {
       const createAndEmailSignInLinkPromise = service.createAndEmailSignInLink(user);
 
-      await expect(createAndEmailSignInLinkPromise).rejects.toThrowError(expectedMessage);
-      await expect(createAndEmailSignInLinkPromise).rejects.toHaveProperty('cause', expectedCause);
+      await expect(() => createAndEmailSignInLinkPromise).rejects.toThrow(expectedMessage);
```
Comment (AlexBramhill) [inline portal-api/src/v1/users/sign-in-link.service.create-and-email-sign-in-link.test.js:361]: good spot!

File: portal-api/src/v1/api.js (hunk 14 above)
Comment (abhi-markan) [reply portal-api/src/v1/api.js]: Have rectified.

File: utils/mock-data-loader/banks/index.js (hunk 13 above)
Comment (abhi-markan) [reply utils/mock-data-loader/banks/index.js:161]: No PR yet, my understanding is this will be have to be reverted in `R2`.

**Review by AlexBramhill — APPROVED** [review]

**Review by AlexBramhill — DISMISSED** [review]

File: README.md [hunk 16]
```diff
@@ -83,12 +83,12 @@ To access GEF locally, use [http://localhost](http://localhost).
 To stop the local environment, simply exit the running terminal and run:
 
 ```shell
-docker-compose down
+docker composedown
```
Comment (AlexBramhill) [inline README.md:86]: typo? `docker compose down`

File: README.md [hunk 17]
```diff
@@ -83,12 +83,12 @@ To access GEF locally, use [http://localhost](http://localhost).
...
 ```
 
-## Different docker-compose files
+## Different docker composefiles
```
Comment (AlexBramhill) [inline README.md:89]: I think `compose files` are two seperate words

File: package.json [hunk 18]
```diff
@@ -28,16 +28,18 @@
   "scripts": {
     "audit:all": "npm audit --if-present",
     "audit:fix:all": "npm audit --fix --if-present",
+    "build": "docker compose build",
```
Comment (AlexBramhill) [inline package.json:31]: I'm not sure on the naming of this as in all other packages we have
`    "build": "rimraf ./public/js && rimraf ./public/css && webpack --mode production --config webpack.prod.config.js",`

which is rather different than building an image?

File: .github/workflows/test.yml [hunk 19]
```diff
@@ -198,7 +198,9 @@ jobs:
 
       - name: Start
         working-directory: ./
-        run: docker-compose -f docker-compose.gha.yml up --build -d
+        run: |
+          docker compose -f docker-compose.gha.yml build
+          docker compose -f docker-compose.gha.yml up -d
```
Comment (AlexBramhill) [inline .github/workflows/test.yml:203]: Why do we split this up by the way? Isn't this the same as before?
https://docs.docker.com/reference/cli/docker/compose/up/

File: package.json [hunk 20]
```diff
@@ -28,16 +28,18 @@
...
-    "dev": "docker-compose up --build",
-    "dev:detach": "npm run dev -- --detach",
     "housekeeping": "npm run update:all && npm run install:all && npm run lint:fix:all && npm run audit:fix:all && npm run spellcheck",
     "postinstall": "husky install",
     "install:all": "npm install --if-present",
     "lint:all": "npm run lint --workspaces --if-present",
     "lint:fix:all": "npm run lint --fix --workspaces --if-present",
     "load": "node utils/mock-data-loader/re-insert-mocks.js",
     "spellcheck": "cspell lint --gitignore --no-must-find-files --unique --no-progress --show-suggestions --color '**/*'",
+    "start": "docker compose up",
```
Comment (AlexBramhill) [inline package.json:40]: Why are we splitting these out of interest?

File: README.md (hunk 16 above)
Comment (abhi-markan) [reply README.md:86]: Fixed

File: .github/workflows/test.yml (hunk 19 above)
Comment (abhi-markan) [reply .github/workflows/test.yml:203]: Have unified

File: package.json (hunk 20 above)
Comment (abhi-markan) [reply package.json:40]: Have unifed.

File: README.md [hunk 21]
```diff
@@ -83,12 +83,12 @@ To access GEF locally, use [http://localhost](http://localhost).
 To stop the local environment, simply exit the running terminal and run:
 
 ```shell
```
Comment (AlexBramhill) [inline README.md:85]: `npm run stop`?

**Review by AlexBramhill — APPROVED** [review]
a nit, that I'm not too worried by

File: README.md (hunk 21 above)
Comment (abhi-markan) [reply README.md:85]: Have added, good point.

**Review by AlexBramhill — APPROVED** [review]
lgtm! great work, lot of code debt tackled

**Pull request merged**